stuff/cron/plagiarism_detector.py

```diff
@@ -45,111 +68,210 @@
                                 s.problem_id, s.verdict, s.guid, s.language
                                 FROM Submissions as s 
                                 INNER JOIN Contests c ON c.problemset_id = s.problemset_id 
                                 WHERE c.contest_id = %s;
                             """
+
+
+class Submission(TypedDict):
+    contest_id: int
+    submission_id: int
+    problemset_id: int
+    problem_id: int
+    verdict: str
+    guid: str
+    language: str
+
+
 INSERT_INTO_PLAGIARISMS = """  
                                 INSERT INTO `Plagiarisms`
                                 (`contest_id`, `score_1`, `score_2` , `submission_id_1` , `submission_id_2`, `contents`)
                                 VALUES
                                 (%s, %s, %s, %s, %s, %s)
                             """
 
-#Constants
+# Constants
 START_RED = "<span class='highlight-red'>"
 START_GREEN = "<span class='highlight-green'>"
 END = "</span>"
 
-SubmissionDownloader = Callable[[str, str], None]
+SubmissionDownloader = Callable[[str, str, str], None]
+
 
 class S3SubmissionDownloader:
     ''' A SubmissionDownloader that can download files from an S3 bucket.  '''
 
     def __init__(self, bucket_name: str = 'omegaup-backup') -> None:
         self._bucket = boto3.client('s3').Bucket(bucket_name)
-    
-    def __call__(self, guid: str, destination_path: str) -> None:
-        self._bucket.download_file(f'omegaup/submissions/{guid[:2]}/{guid[2:]}', destination_path)
-    
+
+    def __call__(self, guid: str, destination_dir: str, language: str) -> None:
+        self._bucket.download_file(
+            f'omegaup/submissions/{guid[:2]}/{guid[2:]}',
+            os.path.join(destination_dir, f'{guid}.{language}'))
+
+
 class LocalSubmissionDownloader:
 
     def __init__(self, dir: str) -> None:
-        self._dir = dir
-    
-    def __call__(self, guid: str, destination_path: str) -> None:
-        
-        shutil.copyfile(os.path.join(self._dir, "test1.cpp"), destination_path)
+        self._dir = os.path.join(SUBMISSION_PATH, dir)
+
+    def __call__(self, guid: str, destination_dir: str, language: str) -> None:
+        shutil.copyfile(os.path.join(self._dir, f'{guid[:2]}/{guid[2:]}'),
+                        os.path.join(destination_dir, guid))
+
+
+"""
+    Function to get the range of lines that are plagiarised. 
+    The length of each list should be even. 
+    [1, 4, 7, 9] So, the ranges are 1-4 and 7-9. 
+"""
+
+
 def get_range(code: Sequence[str]) -> Sequence[int]:
 
-    code_range = []
-    for i in range(0,len(code)):
-        if START_RED in code[i]:
-            code_range.append(i)
-        if( START_GREEN in code[i]):
-            code_range.append(i)
-        if END in code[i]:
-            code_range.append(i)
+    code_range: List[int] = []
+    for line_number, line in enumerate(code):
+
+        #If the color is red, then it will be the same for the entire 'code'.
```

Comment:
```suggestion
        # If the color is red, then it will be the same for the entire 'code'.
```

nit: always use one space between the `#` and the comment.